vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BodyHandlerImpl.java

```diff
@@ -107,10 +108,28 @@ public BodyHandler setDeleteUploadedFilesOnEnd(boolean deleteUploadedFilesOnEnd)
     return this;
   }
 
+  private int getInitialBodyBufferSize(HttpServerRequest request) {
+    String contentLength = request.getHeader(HttpHeaders.CONTENT_LENGTH);
+    if(contentLength == null || contentLength == "") {
+      return DEFAULT_INITIAL_BODY_BUFFER_SIZE;
+    }
+
+    try{
+      long parsedContentLength = Long.parseLong(contentLength);
+      if(parsedContentLength < 0) {
+        return DEFAULT_INITIAL_BODY_BUFFER_SIZE;
+      }
+      return (int)Math.min(parsedContentLength, Integer.MAX_VALUE);
```

Comment:
I'm afraid of this, if someone intentionally sets a large number, it could lead to large allocations and therefore DDoS with OOM

Reply:
I think this is just for getting a value, then this value is used this way:

```
Math.min(bodyLimit, initialBodyBufferSize)
```

that being said I think it should rather return null unknown and be handled as such

Reply:
without preallocation

Reply:
Yes null or -1 would be more sensible to avoid people like me reading just a segment of the code and making wrong assumptions...